tensorboard/webapp/notification_center/_redux/notification_center_effects_test.ts

```diff
@@ -0,0 +1,105 @@
+import {Action, Store} from '@ngrx/store';
+import {MockStore, provideMockStore} from '@ngrx/store/testing';
+import {Subject} from 'rxjs';
+import {State} from '../../app_state';
+import {TBHttpClientTestingModule} from '../../webapp_data_source/tb_http_client_testing';
+import {
+  NotificationCenterDataSource,
+  NotificationCenterResponse,
+} from '../_data_source';
+import * as actions from './notification_center_actions';
+import {
+  NotificationCenterEffects,
+  TEST_ONLY,
+} from './notification_center_effects';
+import {CategoryEnum} from './notification_center_types';
+import {
+  buildDataSourceNotification,
+  provideTestingNotificationCenterDataSource,
+} from './testing';
+
+describe('notification center effects', () => {
+  let dataSource: NotificationCenterDataSource;
+  let effects: NotificationCenterEffects;
+  let store: MockStore<State>;
+  let actions$: Subject<Action>;
+  let actualActions: Action[] = [];
+  let fetchNotificationsSpy: jasmine.Spy;
+  let fetchNotificationSubject: Subject<NotificationCenterResponse>;
+
+  beforeEach(async () => {
+    actions$ = new Subject<Action>();
+    actualActions = [];
+
+    await TestBed.configureTestingModule({
+      imports: [TBHttpClientTestingModule],
+      providers: [
+        provideMockActions(actions$),
+        provideTestingNotificationCenterDataSource(),
+        NotificationCenterEffects,
+        provideMockStore({
+          initialState: {
+            notifications: [],
+            lastReadTimestampInMs: -1,
+          },
+        }),
+      ],
+    }).compileComponents();
+
+    store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
+    spyOn(store, 'dispatch').and.callFake((action: Action) => {
+      actualActions.push(action);
+    });
+    effects = TestBed.inject(NotificationCenterEffects);
+    dataSource = TestBed.inject(NotificationCenterDataSource);
+    effects.initialNotificationFetch$.subscribe();
+
+    fetchNotificationSubject = new Subject();
+    fetchNotificationsSpy = spyOn(
+      dataSource,
+      'fetchNotifications'
+    ).and.returnValue(fetchNotificationSubject);
+  });
+
+  it('fetches initial notifications success', () => {
+    actions$.next(TEST_ONLY.initAction());
+    fetchNotificationSubject.next(buildDataSourceNotification());
+    expect(fetchNotificationsSpy).toHaveBeenCalled();
+    expect(actualActions).toEqual([
+      actions.fetchNotificationsLoaded({
+        notifications: [
+          {
+            category: CategoryEnum.WHATS_NEW,
+            dateInMs: 123,
+            title: 'test title',
+            content: 'random content',
+          },
+        ],
+        error: '',
+      }),
+    ]);
+  });
+
+  it('fetches initial notifications failure', () => {
+    actions$.next(TEST_ONLY.initAction());
+    fetchNotificationSubject.next({error: 'error'});
+    expect(fetchNotificationsSpy).toHaveBeenCalled();
+    expect(actualActions).toEqual([actions.fetchNotificationsFailed()]);
+  });
```

Comment:
`it('fetches initial notifications failure', () =>` -> `it("dispatches failed action when notification fetch failed", () =>`?

Reply:
Thanks for rephrasing the description!